src/main/java/in/twizmwaz/cardinal/module/apply/AppliedModule.java

```diff
@@ -0,0 +1,260 @@
+            filter = filterModule.getFilter(match, applyElement.getAttributeValue(type.filterAttr));
+            if (filter == null) {
+              Located attrLocated = (Located) applyElement.getAttribute(type.filterAttr);
+              errors.add(new ModuleError(this, match.getMap(),
+                  new String[]{"Invalid filter specified for applied region attribute: " + type.filterAttr,
+                      "Element at " + attrLocated.getLine() + ", " + attrLocated.getColumn()}, false));
+              continue;
+            }
+          }
+
+          if (type.filterOnly) {
+            applied.get(match).add(new AppliedRegion(type, region, filter, message, earlyWarning));
+          } else {
+            if (filter == null) {
+              //TODO: always filter doesn't exist yet
+              filter = filterModule.getFilter(match, "always");
+            }
+            if (type.equals(ApplyType.VELOCITY)) {
+              Vector vel = Numbers.getVector(applyElement.getAttributeValue(type.otherAttr));
+              if (vel == null) {
+                Located attrLocated = (Located) applyElement.getAttribute(type.otherAttr);
+                errors.add(new ModuleError(this, match.getMap(),
+                    new String[]{"Invalid vector specified for applied region attribute: " + type.otherAttr,
+                        "Element at " + attrLocated.getLine() + ", " + attrLocated.getColumn()}, false));
+                continue;
+              }
+              applied.get(match).add(new AppliedRegion(region, filter, vel));
+            } else {
+              Kit kit = kitModule.getKit(match, applyElement.getAttributeValue(type.otherAttr));
+              if (kit == null) {
+                Located attrLocated = (Located) applyElement.getAttribute(type.otherAttr);
+                errors.add(new ModuleError(this, match.getMap(),
+                    new String[]{"Invalid kit specified for applied region attribute: " + type.otherAttr,
+                        "Element at " + attrLocated.getLine() + ", " + attrLocated.getColumn()}, false));
+                continue;
+              }
+              applied.get(match).add(new AppliedRegion(type, region, filter, kit));
+            }
+          }
+        }
+      }
+    }
+    return true;
+  }
+
+  public void add(Match match, AppliedRegion appliedRegion) {
+    add(match, appliedRegion, false);
+  }
+
+  public void add(Match match, AppliedRegion appliedRegion, boolean prioritize) {
+    applied.get(match).add(prioritize ? 0 : applied.get(match).size(), appliedRegion);
+  }
+
+  public List<AppliedRegion> get(Match match, ApplyType... types) {
+    List<ApplyType> typeList = Arrays.asList(types);
+    return applied.get(match).stream().filter(appliedRegion ->
+        typeList.contains(appliedRegion.getType())).collect(Collectors.toList());
+  }
+
+  //TODO: Regions shouldn't apply to observers
+  @EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
```

Comment:
Is the best way to logically organize the event handlers to have long chains of ifs? If nothing else, I'd like you to combine any that can be with &&. I recommend using a new line for every condition, however.